### DISCUSSION TITLE
Plans for Binaries like the original

### DISCUSSION THREAD

**Discussion by quorn23** [nzbgetcom/nzbget#132] [General]
Hi Paul,

Quick question, are you planning on providing (updated) binaries like original GH? As in Github releases, when you push any updates? Would make stuff like building an updated container on new release easier etc.

**Comment by paul-chambers** [comment]
I'd have to dig in to understand how hugbug was doing it. Should be possible to automate that. It might be a while before I have time, though.

**Comment by owine** [comment]
For a point of reference, @linuxserver compiles from source for their Docker images. Here is the Docker file which includes the build instructions: https://github.com/linuxserver/docker-nzbget/blob/master/Dockerfile

**Comment by tastyratz** [comment]
For anyone finding this thread elsewhere, Win binaries have been compiled under this forked repository: 
https://github.com/salami-ch/nzbget-win64/releases
I'm not sure if that's helpful to dev, OP, or others searching but I thought it would be worth linking here.

**Comment by mkanet** [comment]
Is there a reason why these two repositories aren't merged into a single multi-platform repository to support Windows and Linux?

![Screenshot 07-22-2023 at 02 19 32 PM](https://github.com/nzbget-ng/nzbget/assets/811008/138feb63-9824-49a6-9638-ba03f81f0bf2)

**Comment by salami-ch** [comment]
I think the most important reason is that there currently isn't any skilled Windows developer willing to contribute. Just because I managed to compile it _somehow_, this doesn't mean my changes are fit to merge into the main project. Even if it doesn't break anything, there are other topics (version numbering, update URLs, x86 support etc.) that should be aligned across the different platforms.

That being said, there is also zero traffic over there and no one is even testing the builds. I suspect most Windows users have already happily switched to SAB.

**Comment by tastyratz** [comment]
> I think the most important reason is that there currently isn't any skilled Windows developer willing to contribute. Just because I managed to compile it _somehow_, this doesn't mean my changes are fit to merge into the main project. Even if it doesn't break anything, there are other topics (version numbering, update URLs, x86 support etc.) that should be aligned across the different platforms.
> 
> That being said, there is also zero traffic over there and no one is even testing the builds. I suspect most Windows users have already happily switched to SAB.

For what it's worth, I've been using your builds from there successfully. I appreciate them and find them helpful. I haven't switched to Sab at all or had much reason to even consider it.

**Reply by mkanet** [reply]
Thanks for the reply.  I actually don't have any reason yet to switch to Sonarr (regardless if I'm using Windows or Linux).  Luckily NZBGet on Windows is rock solid the last 3 years I've used it.  There aren't any features in Sabnzbd currently that NZBGet doesn't have already; at least, not any features I need.

I'll be much more motivated to switch to Sabnzd if there's a new NZB standard I can take advantage of that NZB clients need to support.

**Comment by MauriceW67** [comment]
> That being said, there is also zero traffic over there and no one is even testing the builds. I suspect most Windows users have already happily switched to SAB.

FWIW I've been using your Windows build and it runs just fine. I don't really want to switch to SABnzbd unless I absolutely have to. It's too clunky for me, NZBget looks a lot smoother.

**Comment by zotabee** [comment]
Any chance to have it pushed on Windows Package Manager Community Repository to be able to download and install via winget ? @salami-ch

**Comment by salami-ch** [comment]
That might be a good idea at some point! Right now I still feel the builds are a little too "bleeding edge" to just blindly throw at users without any real testing. The latest build alone took quite some time to fix all the build issues introduced by the updated dependencies, let's see if that breaks anything at runtime :)

**Comment by tastyratz** [comment]
@zortabee Honestly, releases are not rapid enough to warrant the time for a package manager.  At this point I'm updating a couple times a year and probably want more direct involvement than automation anyway.

@salami-ch  I think the NG breakoff and loss of the inbuilt update system fragmented the user base significantly. Stack on that where your windows builds are a fork of a fork... I suspect a ton of people are still using the last stable release none the wiser of the great work still happening, unfortunately.

**Comment by zotabee** [comment]
> @zortabee Honestly, releases are not rapid enough to warrant the time for a package manager. At this point I'm updating a couple times a year and probably want more direct involvement than automation anyway.

Yes I understand but it's low maintenance and I think anybody can submit a repo update as long as the binaries are hosted on github.

It's so convenient I have all my apps getting installed with winget when I do a fresh install also to update them all at once, multiplied by the number of users, that's so practical and win-win for everybody. Also as you said now the risk is that people userbase will be fragmented and I guess a lot don't know about this Windows binaries from salami. We need better visibility and exposure.

**Comment by luckedea** [comment]
Closing this discussion as not vital - we build all necessary binaries/installers and maintain them.
